### PR TITLE
MGDAPI-3795 G104: Audit errors not checked. Error handling added where required. 

### PR DESCRIPTION
**Pull request by Fiona-Waters**
# Issue link
[MGDAPI-3795](https://issues.redhat.com/browse/MGDAPI-3795)

# What
Removing G104 from gosec/exclude command showed a list of 22 unhandled errors. 
Error handling has been added in all cases.

# Verification steps
1. On master branch edit makefile line 543 removing G104
2. Run make gosec/exclude command. This will show 22 issues.
3. Using this [PR,](https://github.com/integr8ly/integreatly-operator/pull/2782) run the gosec/exclude command which will show 0 issues.

### CODE REVIEW

**Comment by Fiona-Waters** [comment]
ok-to-test

**Comment by kryanbeane** [comment]
/lgtm
your changes resolved the gosec issues. All new error handling looks good

**Comment by kryanbeane** [comment]
/lgtm
only change since last lgtm was a broken import

File: test/common/user_dedicated_admin_synced_sso.go [hunk 1]
```diff
@@ -423,29 +423,41 @@ func getKeycloakUserGroups(httpClient *http.Client, host, token string, options
 	return keycloakUserGroups, nil
 }
 
-func cleanUpTestDedicatedAdminUsersSyncedSSO(ctx context.Context, c client.Client) {
+func cleanUpTestDedicatedAdminUsersSyncedSSO(ctx context.Context, t TestingTB, c client.Client) {
 	// Ensure OpenShift user is deleted
-	c.Delete(ctx, testUser)
+	err := c.Delete(ctx, testUser)
+	if err != nil {
+		t.Fatalf("Failed to delete OpenShift user %s, err: %v", testUser.Name, err)
```
Comment (MStokluska) [inline test/common/user_dedicated_admin_synced_sso.go:430]:
```suggestion
		t.Fatalf("failed to delete OpenShift user %s, err: %v", testUser.Name, err)
```

File: test/functional/multiaz_pod_distribution.go [hunk 2]
```diff
@@ -32,7 +32,10 @@ func TestMultiAZPodDistribution(t common.TestingTB, ctx *common.TestingContext)
 	pods := &v1.PodList{}
 	nodes := &v1.NodeList{}
 
-	ctx.Client.List(goctx.TODO(), nodes)
+	err := ctx.Client.List(goctx.TODO(), nodes)
+	if err != nil {
+		t.Fatalf("Error listing nodes: %s", err)
```
Comment (MStokluska) [inline test/functional/multiaz_pod_distribution.go:37]:
```suggestion
		t.Fatalf("error listing nodes: %s", err)
```

**Comment by kryanbeane** [comment]
changes since the last lgtm were just error message changes
/lgtm

**Pull request merged**